### PR TITLE
Use jdk9 

### PR DESCRIPTION
**Pull request by nimearo**
Some minor changes for docker image.

Changes:
- use jdk9
- add memory option for min and max memory
- add jvm options for jdk9
- switched to label v1.4.1.2 instead of HEAD of master iri branch

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
+# procps is very common in build systems, and is a reasonably small package
+RUN apt-get update && apt-get install -y --no-install-recommends \
+		git \
+	&& rm -rf /var/lib/apt/lists/*
+
 
 WORKDIR /iri
 
-RUN git clone -b master https://github.com/iotaledger/iri.git /iri/
+RUN git clone -b v1.4.1.2 https://github.com/iotaledger/iri.git /iri/
```
Comment (bluedigits) [inline Dockerfile]: Bummer, but not possible due to the autobuild functionality of the docker hub project https://hub.docker.com/r/bluedigits/iota-node/
which is triggered by pushes to the official iotaledger/iri project. Pointing to master ensures that the _latest_ version always points to the most recent one.

**Comment by bluedigits** [comment]
Fantastic work, thanks a lot! I'll review it tomorrow.

**One question:**
I've noticed the replacement of the git checkout command which now points to the specific IRI version 1.4.1.2. This would break the docker container https://hub.docker.com/r/bluedigits/iota-node/ because then _latest_ doesn't point to the most recent IRI version anymore. Any ideas?

**Comment by nimearo** [comment]
I am also not completely happy with this solution. Because I always have to update the Dockerfile manually after each release made on the official iri repository. 
In the past there was sometimes a difference between HEAD and latest release label on the master branch. 
That's the reason why I now opted to manually change it because most of the time additional steps like --rescan, --revalidate and/or remove of database are required during update.
I also would like to have automatic detection of latest release on iri master branch.
Nevertheless the version tag can be defined during docker build.

**Comment by Berndinox** [comment]
why is "attachToTangle" limited?

**Comment by bluedigits** [comment]
> In the past there was sometimes a difference between HEAD and latest release label on the master branch.
That's the reason why I now opted to manually change it because most of the time additional steps like --rescan, --revalidate and/or remove of database are required during update.
I also would like to have automatic detection of latest release on iri master branch.
Nevertheless the version tag can be defined during docker build.

Understood.
I talked to Paul Handy and he ensured that from now on IRI releases are always backmerged into master branch. Hence, master always points to the latest release label. I would like to make a small adjustment to the git checkout command, but I of course will accept the rest your pull request.

Thanks for your brilliant work!

**Comment by bluedigits** [comment]
@Berndinox can you open an issue please? :+1:

**Pull request merged**